django/db/models/fields/related.py

```diff
@@ -1318,7 +1346,7 @@ def get_db_prep_lookup(self, lookup_type, value, connection, prepared=False):
 
     def is_hidden(self):
         "Should the related object be hidden?"
-        return self.related_name and self.related_name[-1] == '+'
+        return bool(self.related_name and self.related_name[-1] == '+')
```

Comment:
because is_hidden should always be a boolean flag (related_name can be None). Refactored it to look a bit nicer (removing the boolean cast). let me know if there are problems